core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorSplit.java

```diff
@@ -27,9 +27,18 @@ public interface ConnectorSplit
      * during splits assignment.
      * When false, the split will always be scheduled on one of the addresses returned by {@link #getAddresses()}.
      */
-    boolean isRemotelyAccessible();
+    default boolean isRemotelyAccessible()
+    {
+        return true;
+    }
 
-    List<HostAddress> getAddresses();
+    default List<HostAddress> getAddresses()
+    {
+        if (!isRemotelyAccessible()) {
+            throw new IllegalStateException("getAddresses must be implemented when for splits with isRemotelyAccessible=false");
+        }
+        return List.of();
+    }
 
     Object getInfo();
```

Comment:
no default here?

Reply:
i don't know yet what the default should be. `return this`? 
can be separate?

Reply:
Can be.